tools/autograd/gen_python_functions.py

```diff
@@ -80,6 +80,36 @@
   ${unpack_self}
   ParsedArgs<${max_args}> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
+
+  ${declare_namedtuple_return_types}
+  ${dispatch}
+
+  Py_RETURN_NONE;
+  END_HANDLE_TH_ERRORS
+}
+""")
+
+PY_VARIABLE_FUNCTION_VARARGS_FORWARD_DECLARATION = CodeTemplate("""\
+static PyObject * ${pycname}(PyObject* self_, PyObject* args, PyObject* kwargs);
+""")
+
+# Check for __torch_function__ overrides
+PY_VARIABLE_FUNCTION_VARARGS = CodeTemplate("""\
+static PyObject * ${pycname}(PyObject* self_, PyObject* args, PyObject* kwargs)
+{
+  HANDLE_TH_ERRORS
+  static PythonArgParser parser({
+    ${signatures}
+  }, /*traceable=*/${traceable});
+  ${unpack_self}
+  ParsedArgs<${max_args}> parsed_args;
+  auto r = parser.parse(args, kwargs, parsed_args);
+
+  // Check for __torch_function__ overrides
+  if(r.has_torch_function()){
+    return handle_torch_function(r, args, kwargs, THPVariableFunctions);
+  }
```

Comment:
nit: It's a little irritating we can't just reuse the same template for function/variable, given that the only difference is literally just the `r.has_torch_function()` condition. You could just define that as a separate block and then have a `${check_has_torch_function}` here.

Reply:
Good point! Thanks for the suggestion :)

Reply:
I implemented this suggestion in the last push. I was worried about having a performance impact on methods but I was misunderstanding how the codegen works, you can set the default value to `check_has_torch_function` to the empty string so this doesn't change how methods work at all. It should also make it more straightforward when @hameerabbasi adds `__torch_function__` support for methods in the next pass.